pkgs/development/tools/build-managers/bazel/bazel_7/default.nix

```diff
@@ -117,7 +100,137 @@ let
       unzip
       which
       zip
+      runJdk
+      makeWrapper
+    ];
+
+  # Bootstrap an existing Bazel so we can vendor deps with vendor mode
+  bazel_bootstrap = stdenv.mkDerivation rec {
+    name = "bazel_bootstrap";
+
+    src = if stdenv.hostPlatform.system == "x86_64-linux" then fetchurl {
+      url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel_nojdk-${version}-linux-x86_64";
+      hash = "sha256-05fHtz47OilpOVYawB17VRVEDpycfYTIHBmwYCOyPjI=";
+    } else if stdenv.hostPlatform.system == "aarch64-linux" then fetchurl {
+      url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel_nojdk-${version}-linux-arm64";
+      hash = "sha256-olrlIia/oXWleXp12E+LGXv+F1m4/S4jj/t7p2/xGdM=";
+    } else if stdenv.hostPlatform.system == "x86_64-darwin" then fetchurl {
+      url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel_nojdk-${version}-darwin-x86_64";
+      hash = "sha256-e5RRZTrhzoTMk2NRRl9m2K5+h1/+5mJNW2vF8h0uftc=";
+    } else fetchurl { # stdenv.hostPlatform.system == "aarch64-darwin"
+        url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel_nojdk-${version}-darwin-arm64";
+        hash = "sha256-ciNBdqm+cIrNR1Wu2wt+wGGb6m9i9aE8LhRqL4BJ0Y4=";
+    };
+
+    nativeBuildInputs = defaultShellUtils;
+    buildInputs = [
+      stdenv.cc.cc
+      autoPatchelfHook
+    ];
+
+    dontUnpack = true;
+    dontPatch = true;
+    dontBuild = true;
+    dontStrip = true;
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      install -Dm755 $src $out/bin/bazel
+
+      runHook postInstall
+    '';
+
+    postFixup = ''
+      wrapProgram $out/bin/bazel \
+        --prefix PATH : ${lib.makeBinPath nativeBuildInputs}
+    '';
+
+    meta.sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+  };
+
+  # The bootstrapped Bazel did not work on its own for vendoring, but a wrapped FHS did
+  bazel_fhs = buildFHSEnv {
+    name = "bazel";
+    targetPkgs = _: [ bazel_bootstrap ];
+    runScript = "bazel";
+  };
+
+  # A FOD that vendors the Bazel dependencies using Bazel's new vendor mode.
+  # See https://bazel.build/versions/7.3.0/external/vendor for details.
+  # Note that it may be possible to vendor less than the full set of deps in
+  # the future, as this is approximately 16GB.
+  bazel_deps = stdenv.mkDerivation {
+    name = "bazel_deps";
+    src = fetchurl {
+      url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel-${version}-dist.zip";
+      hash = "sha256-8FAfkMn8dM1pM9vcWeF7jWJy1sCfi448QomFxYlxR8c=";
```

Comment:
this can probably reuse top-level src attribute

Reply:
https://github.com/NixOS/nixpkgs/commit/a0ef387db3403a26b1018db2b8f8c97688ab8a5f